portfolio/src/main/java/com/google/sps/servlets/MarkersServlet.java

```diff
@@ -35,83 +35,83 @@
+      Marker marker = new Marker(lat, lng, userId, id);
+      markers.add(marker);
     }
 
-    /** Deletes Markers */
-    @Override
-    public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
-        if (request.getParameter("id").equals("all")) {
-            Entities entities = new Entities();
-            ArrayList<Long> deletedIds = entities.deleteAll("Marker");
-            System.out.println(deletedIds);
-            Gson gson = new Gson();
-            response.setContentType("application/json;");
-            response.getWriter().println(gson.toJson(deletedIds));
-        } else {
-            long id;
-
-            try {
-                id = Long.parseLong(request.getParameter("id"));
-            } catch (NumberFormatException e) {
-                System.err.println("Could not convert to long");
-                id = -1;
-            }
-
-            if (id > 0) {
-                Entities.deleteSingle(id, "Marker");
-                response.setContentType("text/html;");
-                response.getWriter().println();
-            }
-        }
+    Gson gson = new Gson();
+    response.getWriter().println(gson.toJson(markers));
+  }
+
+  /** Receives and stores a new marker, then sends it. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    String userId = userService.getCurrentUser().getUserId();
+    double lat = Double.parseDouble(request.getParameter("lat"));
+    double lng = Double.parseDouble(request.getParameter("lng"));
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity entity = new Entity("Marker");
+    entity.setProperty("lat", lat);
+    entity.setProperty("lng", lng);
+    entity.setProperty("userId", userId);
+
+    datastore.put(entity);
+    long id = entity.getKey().getId();
+    Marker marker = new Marker(lat, lng, userId, id);
+
+    Gson gson = new Gson();
+    response.getWriter().println(gson.toJson(marker));
+  }
+
+  /** Deletes Markers */
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+
+    if (request.getParameter("id").equals("all")) {
+      Entities entities = new Entities();
+      ArrayList<Long> deletedIds = entities.deleteAll("Marker");
+      System.out.println(deletedIds);
+      Gson gson = new Gson();
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(deletedIds));
+    } else {
+      long id;
+
+      try {
+        id = Long.parseLong(request.getParameter("id"));
+      } catch (NumberFormatException e) {
+        System.err.println("Could not convert to long");
+        id = -1;
+      }
+
+      if (id > 0) {
+        Entities.deleteSingle(id, "Marker");
+        response.setContentType("text/html;");
```

Comment:
We probably ought to have a given endpoint return the same type of response in all cases; so, always JSON, always HTML, etc.

Reply:
Sounds good, I'll change this to respond with JSON as well, thanks!

Would this apply to different HTTP method requests as well? For example in DataServlet.java the type of response for doDelete is text/html but for doGet is JSON - should I also have these return the same response type as well or does it only have to be the same response type within the same request method?

Reply:
As discussed in person, I just mean keeping it consistent with a (URL, method) pair. So, DELETE /markers should always return JSON, whereas it could make sense for POST /markers to always return HTML. So yeah, just keep the response type the same with each servlet handler method.